athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java

```diff
@@ -292,18 +292,43 @@ protected TableName caseInsensitiveTableSearch(Connection connection, final Stri
     public TableName caseInsensitiveTableMaterialViewMatch(Connection connection, final String databaseName,
                                                      final String tableName) throws Exception
     {
-        String resolvedName;
-        PreparedStatement preparedStatement = JDBCUtil.getTableNameQuery(connection, tableName, databaseName);
-        resolvedName = caseInsensitiveNameResolver(preparedStatement, tableName, databaseName);
-
-        if (resolvedName == null) {
-            preparedStatement = getMaterializedViewCaseInsensitive(connection, tableName, databaseName);
-            resolvedName = caseInsensitiveNameResolver(preparedStatement, tableName, databaseName);
-            if (resolvedName == null) {
-                throw new RuntimeException(String.format("During Case Insensitive look up could not find '%s' in Database '%s'", tableName, databaseName));
+        String resolvedSchemaName = databaseName;
+        if (this.getClass().getSimpleName().equals("PostGreSqlMetadataHandler")) {
+            PreparedStatement statement = JDBCUtil.getSchemaNameQuery(connection, databaseName);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    resolvedSchemaName = resultSet.getString("schema_name");
+                }
+                else {
+                    throw new RuntimeException(String.format("During SCHEMA Case Insensitive look up could not find Database '%s'", databaseName));
+                }
+            }
+        }
+        else {
+            // redshift case
```

Comment:
Can we override this method in redshift connector? To avoid having redshift logic in PostGreSql